apps/system/js/lockscreen_window_manager.js

```diff
@@ -89,12 +89,55 @@
     this.startEventListeners();
     this.startObserveSettings();
     this.initElements();
-    this.initWindow();
+    this._fetchFTUStatus().then((status) => {
+      var [exists, enabled] = status;
+      if (!exists) {
+        this.initWindow();
+      } else if (exists && null !== enabled && !enabled){
+        this.initWindow();
+      }
+    });
     Service.register('unlock', this);
     Service.register('lock', this);
     Service.request('registerHierarchy', this);
   };
 
+  /**
+   * Return a Promise with [exists, enabled] result.
+   */
+  LockScreenWindowManager.prototype._fetchFTUStatus = function() {
+
+    // Exists: if in default settings it exists.
+    var lock = navigator.mozSettings.createLock();
+    var setting = lock.get('ftu.manifestURL');
+    var exists = setting.then(() => {
+      if (!setting.result['ftu.manifestURL'] ||
+          '' === setting.result['ftu.manifestURL']) {
+        // Initialize the window only when there is no FTU.
+        return Promise.resolve(false);
+      } else {
+        return Promise.resolve(true);
+      }
+    }).catch((err) => {
+      console.error('Try to get FTU value but it\'s failed');
+      throw err;
+    });
+
+    // Enabled: if it's closed, System would write a value to indicate
+    // it's been launched and now is disabled.
+    var resolve, reject;
+    var enabled = new Promise((res, rej) => {
+      resolve = res;
+      reject = rej;
+    });
+    window.asyncStorage.getItem('ftu.enabled', function(enabled) {
+      // Would be null if it's first time booting.
+      resolve(enabled);
+    });
```

Comment:
I wonder if there should be a convention for a variable storing a promise? Maybe `enabledp` or `pEnabled` ... not sure.

This can stay as-is if we couldn't figure out a better one.